src/coreclr/src/utilcode/pedecoder.cpp

```diff
@@ -290,7 +290,8 @@ CHECK PEDecoder::CheckNTHeaders() const
         // Ideally we would require the layout address to honor the section alignment constraints.
         // However, we do have 8K aligned IL only images which we load on 32 bit platforms. In this
         // case, we can only guarantee OS page alignment (which after all, is good enough.)
-        CHECK(CheckAligned(m_base, GetOsPageSize()));
+        // Also in the case of files embedded within a single-file app, the default alignment for assemblies is 16 bytes.
```

Comment:
This new comment seems to conflict with the previous block or at the least two messages are being given. At a minimum we should remove: "In this case, we can only guarantee OS page alignment (which after all, is good enough.)"

Reply:
The comment tries to say that normally we do have page alignment, even though in some documentation more coarse alignment is specified. And then we have another exception to that for singlefile apps.
I guess the shorter this comment the better.